gammapy/estimators/flux.py

```diff
@@ -38,14 +39,6 @@ class FluxEstimator(Estimator):
         Sigma to use for asymmetric error computation.
     n_sigma_ul : int
         Sigma to use for upper limit computation.
-    backend : str
```

Comment:
Here it seems problematic to assume that users will have to go through passing the full `Fit` object simply to change the `reoptimize` option. In particular this is impossible to configure from the HLI. 
We have to keep the option directly on `__init__`.

Reply:
I think the recommended way of working with the `Estimator` becomes slightly different:
```python3

estimator = FluxEstimator()
estimator.fit.reoptimize = True

result_1 = estimator.run(datasets)

estimator.fit.reoptimize = False
result_2 = estimator.run(datasets)
``` 
However we can maybe find an alternative pattern e.g. based on `kwargs` passed on to `Fit`, or re-exposing the most important parameters (however this is non-ideal, as it is not extensible and arbitrary to a certain degree...)

Reply:
I moved the `reoptimize` option back to `Estimator.__init__`...